lib/gtfs_realtime_viz.ex

```diff
@@ -56,13 +57,75 @@ defmodule GTFSRealtimeViz do
   @spec visualize_diff(term, term, route_opts) :: String.t
   def visualize_diff(group_1, group_2, opts) do
     routes = Map.keys(opts)
-    archive_1 = get_vehicle_archive(group_1, routes)
-    archive_2 = get_vehicle_archive(group_2, routes)
-    [vehicle_archive: Enum.zip(archive_1, archive_2), routes: opts, render_diff?: true]
+    vehicle_archive_1 = get_vehicle_archive(group_1, routes)
+    trip_archive_1 = get_trip_update_archive(group_1, routes)
+    vehicle_archive_2 = get_vehicle_archive(group_2, routes)
+    trip_archive_2 = get_trip_update_archive(group_2, routes)
+    trip_archive = archive_trips(trip_archive_1, trip_archive_2)
+
+    [trip_update_archive: trip_archive, vehicle_archive: Enum.zip(vehicle_archive_1, vehicle_archive_2), routes: opts, render_diff?: true]
     |> gen_html()
     |> Phoenix.HTML.safe_to_string()
   end
 
+  defp archive_trips(trip_set_1, trip_set_2) do
+    archived = Enum.reduce(trip_set_1, %{}, fn {key, value}, acc ->
+      Map.put(acc, key, {value, trip_set_2[key]})
+    end)
+    archived = Enum.reduce(trip_set_2, archived, fn {key, value}, acc ->
+      Map.put(acc, key, {trip_set_1[key], value})
+    end)
+  end
+
+  defp get_trip_update_archive(group, routes) do
+    group
+    |> State.trip_updates
+    |> trips_we_care_about(routes)
+    |> trip_updates_by_stop_id
+  end
+
+  def trips_we_care_about(state, routes) do
+    Enum.map(state,
+      fn {descriptor, update_list} ->
+        filtered_positions = update_list
+        |> Enum.filter(fn trip_update ->
+          trip_update.trip && trip_update.trip.route_id in routes
+        end)
+        {descriptor, filtered_positions}
+      end)
+  end
+
+  defp trip_updates_by_stop_id(state) do
+    Enum.flat_map(state, fn {_descriptor, trip_updates} ->
```

Comment:
Hmm, does this condense one map too far? That is, the `vehicles_by_stop_id` returns a *list* of vehicles-by-stop-id maps, to handle stepping through history. But it looks like at a high level this function flat maps everything together and then yields a single by-stop-id map. Doesn't that consolidate all the historical trip_updates into the single map?